packages/@aws-cdk/aws-codedeploy/lib/deployment-group.ts

```diff
@@ -141,19 +172,84 @@ export class ServerDeploymentGroup extends ServerDeploymentGroupRef {
             assumedBy: new cdk.ServicePrincipal('codedeploy.amazonaws.com'),
             managedPolicyArns: ['arn:aws:iam::aws:policy/service-role/AWSCodeDeployRole'],
         });
+        this._autoScalingGroups = (props && props.autoScalingGroups) || [];
+        this.addCodeDeployAgentInstallUserData(props && props.agentInstallationType);
 
         const resource = new cloudformation.DeploymentGroupResource(this, 'Resource', {
             applicationName: this.application.applicationName,
             deploymentGroupName: props && props.deploymentGroupName,
             serviceRoleArn: this.role.roleArn,
             deploymentConfigName: props && props.deploymentConfig &&
                 props.deploymentConfig.deploymentConfigName,
+            autoScalingGroups: props && props.autoScalingGroups &&
+                props.autoScalingGroups.map(asg => asg.autoScalingGroupName()),
         });
 
         this.deploymentGroupName = resource.ref;
         this.deploymentGroupArn = deploymentGroupName2Arn(this.application.applicationName,
             this.deploymentGroupName);
     }
```

Comment:
Would it be useful to also support `addAutoScalingGroup`?

Reply:
Sure.